bnd-maven-plugin-parent/bnd-maven-plugin/src/main/java/org/bndtools/maven/MavenVersion.java

```diff
@@ -0,0 +1,63 @@
+package org.bndtools.maven;
+
+/**
+ * Represents a Maven version with the ability to translate to a bnd version,
+ * with SNAPSHOT being replaced by bnd's ${tstamp} macro.
+ * 
+ * @author Neil Bartlett <neil.bartlett@paremus.com>
+ */
+public class MavenVersion {
+
+	private static final String SNAPSHOT = "SNAPSHOT";
+	private static final String TSTAMP = "${tstamp}";
+
+	private int major;
+	private int minor;
+	private int micro;
+	private String qualifier;
+	
+	public MavenVersion(String versionString) throws IllegalArgumentException {
```

Comment:
OK, so copy it and add that feature. Otherwise this plugin may fail on a number of versions encountered.

In the gradle plugin, I took the position that if they are deliberately building an OSGi bundle, then the jar version better be a valid OSGi version.